completion.go

```diff
@@ -192,52 +210,75 @@ var availableMessageRoleForO1Models = map[string]struct{}{
 	ChatMessageRoleAssistant: {},
 }
 
-// validateRequestForO1Models checks for deprecated fields of OpenAI models.
-func validateRequestForO1Models(request ChatCompletionRequest) error {
-	if _, found := O1SeriesModels[request.Model]; !found {
+// validateOSeriesRequest checks for fields in requests which are not allowed for o-series models
+// this includes: o1 does not support function calling, or system messages.
+func validateOSeriesRequest(request ChatCompletionRequest) error {
+	_, o1Series := O1SeriesModels[request.Model]
+	_, o3Series := O3SeriesModels[request.Model]
+
+	if !o1Series && !o3Series {
 		return nil
 	}
 
-	if request.MaxTokens > 0 {
-		return ErrO1MaxTokensDeprecated
+	if err := validateBasicParams(request); err != nil {
+		return err
 	}
 
-	// Logprobs: not supported.
-	if request.LogProbs {
-		return ErrO1BetaLimitationsLogprobs
+	if err := validateModelParams(request); err != nil {
+		return err
 	}
 
-	// Message types: user and assistant messages only, system messages are not supported.
-	for _, m := range request.Messages {
-		if _, found := availableMessageRoleForO1Models[m.Role]; !found {
-			return ErrO1BetaLimitationsMessageTypes
+	if o1Series {
+		if err := validateO1Specific(request); err != nil {
+			return err
 		}
 	}
 
-	// Tools: tools, function calling, and response format parameters are not supported
-	for _, t := range request.Tools {
-		if _, found := unsupportedToolsForO1Models[t.Type]; found {
-			return ErrO1BetaLimitationsTools
-		}
+	return nil
+}
+
+func validateBasicParams(request ChatCompletionRequest) error {
```

Comment:
The method name is pretty generic, let's make it's name clearly indicate that it is used only for reasoning models. Maybe even put all the logic related to reasoning models into a separate file (struct?)